mods/easy-emoticon.user.js

```diff
@@ -0,0 +1,150 @@
+// ==UserScript==
+// @name         Kbin Easy Emoticon
+// @namespace    https://github.com/aclist
+// @version      0.6.2
+// @description  Use slash commands for emoticons
+// @author       minnieo
+// @match        https://kbin.social/*
+// @license      MIT
+// ==/UserScript==
+
+let eventListener;
+//lemmy based emoticons
+const emoticons = [
+    '¯\\\\_(ツ)\\_/¯', '( ͡° ͜ʖ ͡°)', '¯\\_( ͡° ͜ʖ ͡°)_/¯', '( ͡° ͜ʖ ͡°)╭∩╮',
+    '( ͡~ ͜ʖ ͡°)', 'ツ', '(͠≖ ͜ʖ͠≖)', '(╯°□°)╯︵ ┻━┻', '┬─┬ノ( º _ ºノ)'
+];
+
+const emoticonsCute = [
+    'ʕ •ᴥ•ʔ', 'ʕっ• ᴥ • ʔっ', '(♡ヮ♡)', '╰(´꒳`)╯',
+    '૮ ˙Ⱉ˙ ა', '( ⸝⸝´꒳`⸝⸝)', '(o/////o " )', '⁄(⁄ ⁄•⁄-⁄•⁄ ⁄)⁄', '( ˶ˆ꒳ˆ˵ )',
+    '(⸝⸝⸝• ω •⸝⸝⸝) ♡', '(   ͡º ꒳ ͡º)', '(˘︶˘).｡.:*♡', '( ˘ ³˘)♥', '(  •̀ - •́  )',
+    '(˵ •̀ ᴗ - ˵ ) ✧', '৻(  •̀ ᗜ •́  ৻)', 'ᕙ( •̀ ᗜ •́ )ᕗ', '(ㅅ´ ˘ `)', '(╥﹏╥)',
+    '( • ᴖ • ｡ )', '◝(ᵔᵕᵔ)◜', '♡⟡˙⋆', ' ⋆˙⟡♡', '❤︎', '♡', '★', '✿',
+    '(｡•́︿•̀｡)', '(⸝⸝ᵕᴗᵕ⸝⸝)', '(ﾉ^ヮ^)ﾉ', 'ᐠ( ᐛ )ᐟ', '(¬_¬")', '( ´･･)ﾉ(._.`)',
+    'ಠ_ಠ', 'ᕕ( ᐛ )ᕗ', '（◞‸◟）', '( っ- ‸ - c)', ' ( ͡°⁄ ⁄ ͜⁄ ⁄ʖ ⁄ ⁄ ͡°)',
+    '╮（￣▽￣）╭', '༼ つ ◕_◕ ༽つ', '(งツ)ว', '( •̯́ ^ •̯̀)', '✧⁺⸜(●′▾‵●)⸝⁺✧',
+    '( ་ ⍸ ་ )', '✽-(ˆ▽ˆ)/✽ ✽\(ˆ▽ˆ)-✽', '(•⌓• )⁼³₌₃', '(•-• )', '(ó﹏ò｡)',
+    '(¬‿¬)', '(ﾉ◕ヮ◕)ﾉ*:･ﾟ✧', '(/▽＼)', '(。。\\)', '눈_눈', '(￣▽￣)"',
+    '( ￢ω￢)', '╮(╯-╰)╭', '(๑╹ᆺ╹)', '(≖_≖ )', 'ヽ༼ຈل͜ຈ༽ﾉ', 'ヽ༼° ل͜ °༽ﾉ',
+    'ヽ༼⊙ل͜⊙༽ﾉ', 'ヽ༼≖ل͜≖༽ﾉ', 'ヽ༼ ・ ل͜ ・ ༽ﾉ', 'ヽ༼◉ل͜◉༽ﾉ', 'ヽ༼ ºلº ༽ﾉ', 
+    '(︶^︶)', '(◉ω◉)', '(・へ・)', '(　´_ゝ`)', '\'ㅅ\'', '(_ _;)', '(⚆_⚆)',
+    '( ´▽`）o自自o（\´▽` )', '(　´▽`)ﾉ(´･ω･`)', '♪┏(・o･)┛♪', 'щ(ﾟДﾟщ)', '( ﾟヮﾟ)...'
+];
+
+function emoticonGen() {
+    eventListener = (e) => {
+        if (e.target.tagName === 'TEXTAREA') {
+            emoticonMake(e.target);
+            console.log('working');
+        }
+    };
+    document.addEventListener('input', eventListener)
+}
+
+function emoticonMake(param) {
+    let text = param.value;
+    text = text
```

Comment:
Instead of using all of these `.replace()`s, you can create a dictionary of commands and emoticons, and use a for loop to replace all of the commands with their respective emoticons. I'm not 100% sure if this would work properly but this should be a good start for cleaning this up and making it easier to add more emoticons:

```
// Replace the emoticons array with a dictionary
const emoticons = {
    /\/shrug/ = '¯\\\\_(ツ)\\_/¯',
    /\/lenny/ = '( ͡° ͜ʖ ͡°)',
    /\/lenshrug/ = '¯\\_( ͡° ͜ʖ ͡°)_/¯',
    // etc.
};

function emoticonMake(param) {
    let text = param.value;

    for (const [command, emoticon] of Object.entries(emoticons)) {
        text.replace(command, `${emoticon} `);
    }

    param.value = text;
}

Reply:
woooah that is cleaner. on it! thanks for helping a beginner learn new things ✧⁺⸜(●′▾‵●)⸝⁺✧'